esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2162,6 +2162,572 @@ bool Cws_accessEx::onResourcePermissions(IEspContext &context, IEspResourcePermi
+        resp.setResult("Successfully deleted a view.");
+    }
+    catch (IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onQueryViewColumns(IEspContext &context, IEspQueryViewColumnsRequest &req, IEspQueryViewColumnsResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+        
+        IArrayOf<IEspViewColumn> viewColumns;
+        StringArray files, columns;
+
+        try
+        {
+            secmgr->queryViewColumns(req.getViewname(), files, columns);
+
+            ForEachItemIn(i, files)
+            {
+                Owned<IEspViewColumn> oneViewColumn = createViewColumn();
+                oneViewColumn->setViewname(req.getViewname());
+                oneViewColumn->setFilename(files.item(i));
+                oneViewColumn->setColumnname(columns.item(i));
+                viewColumns.append(*oneViewColumn.getLink());
+            }
+        }
+        catch (IException* e)
+        {
+            StringBuffer failedMsg("Failed: ");
+            StringBuffer eMsg;
+            resp.setSuccess(false);
+            resp.setResult(failedMsg.append(e->errorMessage(eMsg)));
+            return false;
+        }
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully queried all view columns.");
+        resp.setViewcolumns(viewColumns);
+    }
+    catch (IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onAddViewColumn(IEspContext &context, IEspAddViewColumnRequest &req, IEspAddViewColumnResponse &resp)
+{
+    try
+    {
+        CLdapSecManager* secmgr = queryLDAPSecurityManager(context);
+        
+        if(secmgr == NULL)
+            throw MakeStringException(ECLWATCH_INVALID_SEC_MANAGER, MSG_SEC_MANAGER_IS_NULL);
+
+        checkUser(context);
+
+        try
+        {
+            StringArray files, columns;
+
+            files.append(req.getFilename());
+            columns.append(req.getColumnname());
+
+            secmgr->addViewColumns(req.getViewname(), files, columns);
+        }
+        catch (IException* e)
+        {
+            StringBuffer failedMsg("Failed: ");
+            StringBuffer eMsg;
+            resp.setSuccess(false);
+            resp.setResult(failedMsg.append(e->errorMessage(eMsg)));
+            return false;
+        }
+
+        resp.setSuccess(true);
+        resp.setResult("Successfully added a view column.");
+    }
+    catch (IException* e)
+    {
+        FORWARDEXCEPTION(context, e, ECLWATCH_INTERNAL_ERROR);
+    }
+
+    return true;
+}
+
+bool Cws_accessEx::onDeleteViewColumn(IEspContext &context, IEspDeleteViewColumnRequest &req, IEspDeleteViewColumnResponse &resp)
+{
+    try
```

Comment:
Btw if I were working on this myself I would refactor both queryLDAPSecurityManager and checkuser, because they are both querying the secmgr and doing some checking.  For example having an inner layer of checkuser where you could pass the secmgr since you've already resolved it.

Besides being repetitive it's a shame there is so much overlap between the queryLDAPSecurityManager, the null check, and the checkuser.  i.e. checkuser gets the LDAPSecurityManager and throws an exception if its null.

But @ghalliday you may not consider that critical enough to prevent merging.